File: scripts/download-posters.js
const http  = require('http');
const fs    = require('fs');
const path  = require('path');

const API_KEY    = '4zbcbwxse5pe84q2x2efevj5';
const BASE_URL   = 'http://developer.tmsimg.com/';
const movies     = require('../src/data/movies.json');
const postersDir = path.join(__dirname, '../public/posters');

if (!fs.existsSync(postersDir)) fs.mkdirSync(postersDir, { recursive: true });

function downloadImage(url, dest) {
  return new Promise((resolve, reject) => {
    const file = fs.createWriteStream(dest);
    http.get(url, res => {
      if (res.statusCode === 200) {
        res.pipe(file);
        file.on('finish', () => { file.close(); resolve(true); });
      } else {
        file.close();
        fs.unlink(dest, () => {});
        reject(`HTTP ${res.statusCode}`);
      }
    }).on('error', err => {
      file.close();
      fs.unlink(dest, () => {});
      reject(err.message);
    });
  });
}

async function main() {
  let downloaded = 0, skipped = 0, failed = 0;

  for (const movie of movies) {
    const rawUri = movie.preferredImage?.uri;
    if (!rawUri) { skipped++; continue; }

    const uriPath  = rawUri.split('?')[0];           // assets/p31207784_v_v11_ad.jpg
    const filename = uriPath.split('/').pop();        // p31207784_v_v11_ad.jpg
    const dest     = path.join(postersDir, filename);

    if (fs.existsSync(dest)) {
      process.stdout.write(`  skip  ${filename}\n`);
      skipped++;
      continue;
    }

    const url = `${BASE_URL}${rawUri}&api_key=${API_KEY}`;

    try {
      await downloadImage(url, dest);
      process.stdout.write(`  ✓     ${filename}\n`);
      downloaded++;
    } catch (err) {
      process.stdout.write(`  ✗     ${filename} — ${err}\n`);
      failed++;
    }
  }

  console.log(`\nDone — downloaded: ${downloaded}  skipped: ${skipped}  failed: ${failed}`);
}

main();

File: scripts/download-posters.py
import json
import os
import urllib.request
import urllib.error

API_KEY    = '4zbcbwxse5pe84q2x2efevj5'
BASE_URL   = 'http://developer.tmsimg.com/'
SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
MOVIES_JSON = os.path.join(SCRIPT_DIR, '..', 'src', 'data', 'movies.json')
POSTERS_DIR = os.path.join(SCRIPT_DIR, '..', 'public', 'posters')

os.makedirs(POSTERS_DIR, exist_ok=True)

with open(MOVIES_JSON, encoding='utf-8') as f:
    movies = json.load(f)

downloaded = skipped = failed = 0

for movie in movies:
    img = movie.get('preferredImage', {})
    raw_uri = img.get('uri', '') if img else ''
    if not raw_uri:
        skipped += 1
        continue

    uri_path = raw_uri.split('?')[0]          # assets/p31207784_v_v11_ad.jpg
    filename = uri_path.split('/')[-1]         # p31207784_v_v11_ad.jpg
    dest     = os.path.join(POSTERS_DIR, filename)

    if os.path.exists(dest):
        print(f'  skip  {filename}', flush=True)
        skipped += 1
        continue

    url = f'{BASE_URL}{raw_uri}&api_key={API_KEY}'

    try:
        req = urllib.request.Request(url, headers={'Referer': ''})
        with urllib.request.urlopen(req, timeout=15) as resp, open(dest, 'wb') as out:
            out.write(resp.read())
        print(f'  OK    {filename}')
        downloaded += 1
    except Exception as e:
        print(f'  FAIL  {filename} -- {e}')
        failed += 1

print(f'\nDone — downloaded: {downloaded}  skipped: {skipped}  failed: {failed}')
